sensor/common/networkflow/service/service_impl.go

```diff
@@ -180,6 +194,15 @@ func (s *serviceImpl) runRecv(stream sensor.NetworkConnectionInfoService_PushNet
 			errC <- err
 			return
 		}
+		if s.writer != nil {
+			if data, err := msg.Marshal(); err == nil {
+				if _, err := s.writer.Write(data); err != nil {
+					log.Warnf("Error writing msg: %v", err)
+				}
+			} else {
+				log.Warnf("Error marshalling  msg: %v", err)
```

Comment:
I didn't want to continue the loop if `Marshaling` fails for whatever reason. These writers are only for testing/debugging and I thought it'd be better to send the messages regardless.

I could flatten the nested `if` like:

```golang
if s.writer != nil {
	data, err := msg.Marshal()
    if err != nil {
		log.Warnf("Error marshalling  msg: %v", err)
	}
	if err == nil {
		if _, err := s.writer.Write(data); err != nil {
			log.Warnf("Error writing msg: %v", err)
		}
	}
```
Or maybe separate the logic into a different function. WDYT?

Reply:
IIUC, `runRecv` runs as a goroutine, so we don't really want it exiting with a `return`, but instead just skipping the rest of the loop,correct?
Wouldn't the rest of the loop be skipped with a `continue` statement, with execution returning back to line 192? 
Given that would work, I had something like this in mind (untested):

```go
if s.writer == nil { continue }
data, err := msg.Marshal()
if err != nil { 
  log.Warnf("Error marshalling msg: %v", err)
  continue
}
if _, err := s.writer.Write(data); err != nil { 
  log.Warnf("Error writing msg: %v", err)
}
```

Reply:
I don't think the first `continue` in your snippet should be there. The `writer` is an optional field that in most cases would be `nil`. By continuing there we will skip the `select` statement, thus skipping the processing of this event.

The second `continue` is the one that is debatable. Since the use case of this writer-block is a test/debugging only thing, I didn't want to skip the `select`. If the marshalling was to fail for whatever reason, I'd rather have the code log the error but keep the production code the same. Does that make sense?